web-admin/src/features/dashboards/query-mappers/getDashboardFromComparisonRequest.ts

```diff
@@ -0,0 +1,84 @@
+import { getSelectedTimeRange } from "@rilldata/web-admin/features/dashboards/query-mappers/utils";
+import type { QueryMapperArgs } from "@rilldata/web-admin/features/dashboards/query-mappers/types";
+import { getSortType } from "@rilldata/web-common/features/dashboards/leaderboard/leaderboard-utils";
+import { SortDirection } from "@rilldata/web-common/features/dashboards/proto-state/derived-types";
+import { TimeComparisonOption } from "@rilldata/web-common/lib/time/types";
+import { DashboardState_ActivePage } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import {
+  V1MetricsViewComparisonMeasureType,
+  type V1MetricsViewComparisonRequest,
+} from "@rilldata/web-common/runtime-client";
+
+export async function getDashboardFromComparisonRequest({
+  req,
+  dashboard,
+  metricsView,
+  timeRangeSummary,
+  executionTime,
+}: QueryMapperArgs<V1MetricsViewComparisonRequest>) {
+  if (req.where) dashboard.whereFilter = req.where;
+
+  if (req.timeRange) {
+    dashboard.selectedTimeRange = getSelectedTimeRange(
+      req.timeRange,
+      timeRangeSummary,
+      req.timeRange.isoDuration,
+      executionTime,
+    );
+  }
+
+  if (req.timeRange?.timeZone) {
+    dashboard.selectedTimezone = req.timeRange?.timeZone || "UTC";
+  }
+
+  if (req.comparisonTimeRange) {
+    if (
+      !req.comparisonTimeRange.isoOffset &&
+      req.comparisonTimeRange.isoDuration
+    ) {
+      dashboard.selectedComparisonTimeRange = {
+        name: TimeComparisonOption.CONTIGUOUS,
+        start: undefined as unknown as Date,
```

Comment:
**Issue**: The use of 'undefined as unknown as Date' is potentially unsafe and unclear. This type assertion bypasses TypeScript's type safety, leading to runtime errors if the object is accessed as a Date. <br> **Fix**: It's recommended to handle this scenario more safely by either initializing dates to a valid Date object or by modifying the logic to avoid the need for a placeholder date. If a placeholder is necessary, consider using 'null' and adjusting the type definition to 'Date | null'. <br> **Code Suggestion**: 
 ```
 -        start: undefined as unknown as Date,
 -        end: undefined as unknown as Date,
 +        start: null,
 +        end: null,
 ```

Reply:
This is an issue with our type definition for time range. Not in the scope of this PR to fix